src/action.c

```diff
@@ -1109,9 +1109,18 @@ actions_run(struct view *activator, struct server *server,
 			break;
 		case ACTION_TYPE_TOGGLE_TEARING:
 			if (view) {
-				view->tearing_hint = !view->tearing_hint;
-				wlr_log(WLR_DEBUG, "tearing %sabled",
-					view->tearing_hint ? "en" : "dis");
+				switch (view->force_tearing) {
+				case LAB_STATE_UNSPECIFIED:
```

Comment:
I think this behavior would be more pleasant if the UNSPECIFIED behavior did

    view->force_tearing = !get_tearing_preference(...)

where `get_tearing_public` from `output.c` will have to be made public and renamed accordingly (*e.g.*, `output_get_tearing_allowance`). This truly toggles even from an indeterminate state.

Reply:
yes, implemented like you suggested and this indeed works nicely.